File: server/php/src/Infrastructure/Mappers/PaymentDetails/PaymentProduct840CustomerAccountMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentProduct840CustomerAccount as PaymentProduct840CustomerAccountDto;
use OnlinePayments\Sdk\Domain\PaymentProduct840CustomerAccount as PaymentProduct840CustomerAccountSdk;

final class PaymentProduct840CustomerAccountMapper
{
    public static function mapFromSdkResponse(?PaymentProduct840CustomerAccountSdk $response): ?PaymentProduct840CustomerAccountDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new PaymentProduct840CustomerAccountDto();
        $dto->accountId = $response?->accountId ?? null;
        $dto->companyName = $response?->companyName ?? null;
        $dto->countryCode = $response?->countryCode ?? null;
        $dto->firstName = $response?->firstName ?? null;
        $dto->customerAccountStatus = $response?->customerAccountStatus ?? null;
        $dto->customerAddressStatus = $response?->customerAddressStatus ?? null;
        $dto->payerId = $response?->payerId ?? null;
        $dto->surname = $response?->surname ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/PaymentProduct840CustomerAccountMapper.cs
using PaymentProduct840CustomerAccountSdk = OnlinePayments.Sdk.Domain.PaymentProduct840CustomerAccount;
using PaymentProduct840CustomerAccountDto = Business.Domain.Payments.PaymentDetails.PaymentProduct840CustomerAccount;

namespace Infrastructure.Mappers.PaymentDetails;

public static class PaymentProduct840CustomerAccountMapper
{
    public static PaymentProduct840CustomerAccountDto Map(PaymentProduct840CustomerAccountSdk? response)
    {
        return new()
        {
            AccountId = response?.AccountId,
            CompanyName = response?.CompanyName,
            CountryCode = response?.CountryCode,
            FirstName = response?.FirstName,
            CustomerAccountStatus = response?.CustomerAccountStatus,
            CustomerAddressStatus = response?.CustomerAddressStatus,
            PayerId = response?.PayerId,
            Surname = response?.Surname
        };
    }
}